### ISSUE TITLE
[BUG] Update cmake version as upcoming deprecation

### ISSUE THREAD

**Issue by chryswoods** [OpenBioSim/sire#184]
We currently set `cmake_minimum_required` to `3.3.0`. CMake support for <3.5 is deprecated and will soon be dropped. We should update to a newer version for 2024.2.0.

As part of this, it may also be worth reviewing some of our pins. For example, the packaging pin caused issue with boa and is being removed in 2024.1.0. Also, there is a pin for compiler 12.3.0 on Linux. We should check if the issues we saw for newer gcc have been resolved.

Also, python 3.12 support looks like it is almost there for all our dependencies. We should see if we can move to 3.10, 3.11, 3.12 for 2024.2.0.

**Additional context**
I've added this issue to remind myself to explore this as part of the next major version.

**Comment by lohedges** [comment]
Yes, that's a good idea. I believe the gcc pin was for RDKit. It's probably worth asking Exs if we can unpin or update the RDKit version as it's quite old now. I don't think RDKit was the issue as of itself, rather the environment resolution or a dependency that was pulled in.

**Comment by lohedges** [comment]
It actually looks like we only set a minimum version. Perhaps they retained the pin for their internal build so this is hopefully no longer needed.
